e2e/cypress/integration/plugins/demo_plugin/webhook_spec.js

```diff
@@ -0,0 +1,232 @@
+        cy.wait(TIMEOUTS.TWO_SEC);
+        cy.postMessage('some message');
+
+        // # Get last post ID
+        cy.getLastPostId().then((postID) => {
+            // # click  dot menu button
+            cy.clickPostDotMenu();
+
+            // # click edit post
+            cy.get(`#edit_post_${postID}`).click();
+
+            // # Edit message to 'some message add'
+            cy.get('#edit_textbox').
+                should('be.visible').
+                and('be.focused').
+                wait(TIMEOUTS.HALF_SEC).
+                type(' add');
+
+            // # Click button Edit
+            cy.get('#editButton').click();
+
+            // # Open demo plugin channel
+            cy.visit(`/${testTeam.name}/channels/demo_plugin`);
+            cy.wait(TIMEOUTS.TWO_SEC);
+
+            // * Verify event posted in the channel
+            cy.findAllByTestId('postView').contains(`MessageHasBeenUpdated: @${testUser.username}, ~Town Square`);
+        });
+    });
+
+    it('MM-T2408/7 - User adds a reaction to a message Webhook event', () => {
+        // # Post message
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+        cy.postMessage('some message');
+
+        cy.getLastPostId().then((postId) => {
+            // # Click the add reaction icon
+            cy.clickPostReactionIcon(postId);
+
+            // # Choose "slightly_frowning_face" emoji
+            // delaying 500ms in case of lag
+            // eslint-disable-next-line cypress/no-unnecessary-waiting
+            cy.get('.emoji-picker__items #emoji-1f641').wait(500).click();
+
+            // # Open demo plugin channel
+            cy.visit(`/${testTeam.name}/channels/demo_plugin`);
+            cy.wait(TIMEOUTS.TWO_SEC);
+
+            // * Verify event posted in the channel
+            cy.findAllByTestId('postView').contains(`ReactionHasBeenAdded: @${testUser.username}, `);
+            cy.findAllByTestId('postView').contains(':slightly_frowning_face:');
+        });
+    });
+
+    it('MM-T2408/8/9 - ​User joined the team Webhook event', () => {
+        // # Add user to the team
+        cy.apiAdminLogin(admin);
+        cy.apiAddUserToTeam(newTeam.id, testUser.id);
+
+        // # Open demo plugin channel
+        cy.visit(`/${newTeam.name}/channels/demo_plugin`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+
+        // * Verify event is posted in the channel
+        cy.findAllByTestId('postView').contains(`UserHasJoinedTeam: @${testUser.username}`);
+
+        // # Delete user from team
+        cy.apiDeleteUserFromTeam(newTeam.id, testUser.id);
+
+        // # Open demo plugin channel
+        cy.visit(`/${newTeam.name}/channels/demo_plugin`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+
+        // * Verify event is posted in the channel
+        cy.findAllByTestId('postView').contains(`UserHasLeftTeam: @${testUser.username}`);
+    });
+
+    it('MM-T2408/10 - ​User user login Webhook event', () => {
+        // # Login
```

Comment:
Not sure what the exact pattern is , but I've seen cy.logout() being called before. Also in the following step MM-T2408/11 , we login as admin (to compensate for logging-in as test user before?) - if yes, prob a more consistent approach would be to add a beforeEach() call where we always login as admin.

Reply:
Not all the test cases need admin login. In the 11 we login as admin because we want to create another user. 
And in 10 we login as a testuser to trigger a login webhook event. Logout is not needed. 
Tests are self-sufficient and can be run in any order or parallel.